include/dali/core/geom/vec.h

```diff
@@ -485,46 +485,48 @@ clamp(const vec<N, T> &in, const vec<N, T> &lo, const vec<N, T> &hi) {
   IMPL_VEC_ELEMENTWISE(clamp(in[i], lo[i], hi[i]));
 }
 
-#ifdef __CUDA_ARCH__
+#if (defined(__NVCC__) && defined(__CUDA_ARCH__)) || (defined(__clang__) && defined(__CUDA__))
 template <int N>
-__device__ vec<N> floor(const vec<N> &a) {
+constexpr DALI_DEVICE vec<N> floor(const vec<N> &a) {
```

Comment:
I see no particular benefit of using DALI_DEVICE over __device__. It's compiled only in CUDA and therefore the keyword will always be available.

Reply:
Oh, I see the point now, we need DALI_HOST_DEV and DALI_HOST (or provide `__host__` and `__device__` without CUDA headers on our own). So the `DALI_DEVICE` is more or less for completeness.

Reply:
> So the DALI_DEVICE is more or less for completeness.

That's right.